From b8a496a3ed730fccf842592baf80d3c81cdb5107 Mon Sep 17 00:00:00 2001
From: auto commit <unknown@unknown>
Date: Fri, 24 Oct 2008 12:22:35 -0700
Subject: [PATCH] cavium: sg->page is gone

There is no longer a sg->page -- it is replaced with sg_page(sg)

Signed-off-by: Tomaso Paoletti <tpaoletti@caviumnetworks.com>
Signed-off-by: Paul Gortmaker <Paul.Gortmaker@windriver.com>
---
 arch/mips/cavium-octeon/dma-octeon.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/arch/mips/cavium-octeon/dma-octeon.c b/arch/mips/cavium-octeon/dma-octeon.c
index 3cfed46..5b4b2d2 100644
--- a/arch/mips/cavium-octeon/dma-octeon.c
+++ b/arch/mips/cavium-octeon/dma-octeon.c
@@ -17,6 +17,7 @@
 #include <linux/string.h>
 #include <linux/dma-mapping.h>
 #include <linux/platform_device.h>
+#include <linux/scatterlist.h>
 
 #include <asm/cache.h>
 #include <asm/io.h>
@@ -365,7 +366,7 @@ int dma_map_sg(struct device *dev, struct scatterlist *sg, int nents,
 	int i;
 	for (i = 0; i < nents; i++, sg++)
 		sg->dma_address =
-			dma_map_page(dev, sg->page, sg->offset, sg->length,
+			dma_map_page(dev, sg_page(sg), sg->offset, sg->length,
 				     direction);
 
 	return nents;
-- 
1.5.5.1

